### PR TITLE
ISPN-12458 Index and Query stats REST API

### PR DESCRIPTION
**Pull request by karesti**
* Move the Search code to a Search dedicated file
* Remove deprecated query and index stats calls
* Display the new index and query stats

https://issues.redhat.com/browse/ISPN-12458

### CODE REVIEW

File: src/services/searchService.ts [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+    maxResults: number,
+    offset: number
+  ): Promise<Either<ActionResponse, SearchResut>> {
+    return this.utils
+      .restCall(
+        this.endpoint +
+          encodeURIComponent(cacheName) +
+          '?action=search' +
+          '&query=' +
+          query +
```
Comment (danberindei) [inline src/services/searchService.ts:38]: Shouldn't the query be encoded like the cache name?

**Review by danberindei — COMMENTED** [review]
I had a quick look, but I'd rather someone who knows more about query merges it

**Comment by karesti** [comment]
I'm going to merge it myself @danberindei because nobody will check any further at this point. if you want to merge it I would love it as well

**Comment by danberindei** [comment]
Good call Katia, I couldn't add anything meaningful.

**Pull request merged**